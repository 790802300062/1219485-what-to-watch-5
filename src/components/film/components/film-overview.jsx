import React from "react";
import PropTypes from "prop-types";
import countRateDescription from "../../../utils/film";
import {filmValidator} from "../../../props-validators/props-validators";

const FilmOverview = (props) => {
  const {promoFilm} = props;

  return (
    <>
      <div className="movie-rating">
        <div className="movie-rating__score">{promoFilm.rate}</div>
        <p className="movie-rating__meta">
          <span className="movie-rating__level">{countRateDescription(promoFilm.rate)}</span>
          <span className="movie-rating__count">{promoFilm.ratings} ratings</span>
        </p>
      </div>
      <div className="movie-card__text">
        <p>{promoFilm.description}</p>

        <p className="movie-card__director"><strong>Director: {promoFilm.director}</strong></p>

        <p className="movie-card__starring"><strong>Starring: {promoFilm.starring}</strong></p>
      </div>
    </>
  );
};

FilmOverview.propTypes = {
  promoFilm: PropTypes.shape(filmValidator)
};

export default FilmOverview;
