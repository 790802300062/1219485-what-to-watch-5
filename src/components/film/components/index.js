import FilmOverview from "./film-overview";
import FilmDetails from "./film-details";
import FilmReviews from "./film-reviews";

export default {
  FilmOverview,
  FilmDetails,
  FilmReviews
};
