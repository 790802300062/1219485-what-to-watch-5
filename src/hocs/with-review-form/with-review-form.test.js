import React from "react";
import renderer from "react-test-renderer";
import withReviewForm from "./with-review-form";
import {MockComponentWithChildren} from "../../test-data/components";

const MockComponentWrapped = withReviewForm(MockComponentWithChildren);

it(`renders withReviewForm`, () => {
  const tree = renderer
    .create(
        <MockComponentWrapped filmId={1}>
          <React.Fragment />
        </MockComponentWrapped>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
