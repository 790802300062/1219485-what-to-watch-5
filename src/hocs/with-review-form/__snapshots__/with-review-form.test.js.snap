// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders withReviewForm 1`] = `<div />`;
